### PR TITLE
Ants ants ants

### PR DESCRIPTION
**Pull request by fjebaker**
Pushed basic modules for creatures, including random walk algorithm and image rendering.

Merged with @jwiggins world branch, so that it renders 10 `Ant`.

Required some 60x90 image `assets/ant.jpg` to render. Currently no error handling, so will fail without image. I've sent an image to the discord as a placeholder until we get proper assets.

### CODE REVIEW

**Comment by fjebaker** [comment]
I added common table idioms to `src/utils.fnl`, the sort of stuff it took me less time to implement myself that to search in the docs to see if they already exist.

It's a jam after all ;)

**Comment by fjebaker** [comment]
Also `src/creature.fnl` now "extends" `src/worldobject.fnl`, adding movement callback functions.

**Comment by fjebaker** [comment]
So I've started working on some collision logic, and I'm planning on probably treating collisions "between creatures" and "creatures and objects" in the world differently. At the moment there's a few issues where every so often a floating division means a creature effectively teleports away, but I'm sure I can iron that out after some sleep.

@jwiggins I'll let you review / merge this branch when you wake up. I've posted temporary asset files to the discord so you can test it.

**Comment by fjebaker** [comment]
Pushed an updated `fmain.fnl` and the changes I made to `src/creeps/ant.fnl` for the hitboxes.

File: src/utils.fnl
Comment (jwiggins) [inline src/utils.fnl:19]: That's just a shallow copy because you're not recursing into values.

File: world.fnl
Comment (jwiggins) [inline world.fnl:24]: Nice

File: world.fnl
Comment (jwiggins) [inline world.fnl:27]: We can delete the `hsv` function now too

File: src/creature.fnl
Comment (jwiggins) [inline src/creature.fnl:66]: There's a [whole physics engine](https://love2d.org/wiki/love.physics) inside LÖVE already. We should definitely use that instead of rolling our own.

File: src/collisions.fnl
Comment (jwiggins) [inline src/collisions.fnl:5]: The big optimization is to use `love.physics`, which (without checking) is C/C++ code that should have adequate performance.

**Review by jwiggins — APPROVED** [review]
I'm gonna merge this so I can integrate and move things along.

**Pull request merged**
